src/handlers.utils.js

```diff
@@ -0,0 +1,56 @@
+const R = require('ramda');
+const SlackTemplate = require('claudia-bot-builder').slackTemplate;
+const { InvalidUserInputError, InvalidCallbackActionError } = require('./handlers.errors');
+
+const validateSendingUser = (userRepository, handler) => async (message, ...extra) => {
+    const { sender } = message;
+    const userPromise = userRepository.getUser(sender)
+        .then(user => {
+            if (!user) {
+                throw new InvalidUserInputError(
+                    new SlackTemplate('I don\'t know who you are. :crying_cat_face:')
+                        .replaceOriginal(false)
+                        .get(),
+                );
+            }
+            return user;
+        });
+    return await handler(message, ...extra, userPromise);
```

Comment:
I'm not even sure that this `...extra` trick will work, but the idea was to be able to combine / pipe several `validate*` methods.

Reply:
How about something like:
```
    const commandHandlers = {
        tournaments: [
          transformReqBody,
          validateUser,
          validateSomeOtherThing, 
          handlers.listOpenTournaments
        ],
        whoami: [handlers.showCurrentUser],
    };
```
Then just pipe/pipeWith the array of functions